webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/checks/migration/CheckTableReportsExecutionTz.scala

```diff
@@ -0,0 +1,120 @@
+ * documentation that, without modification of the Source Code, enables
+ * supplementary functions or services in addition to those offered by
+ * the Software.
+ *
+ * Rudder is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+import bootstrap.liftweb.*
+import com.normation.errors.IOResult
+import com.normation.rudder.db.Doobie
+import com.normation.zio.*
+import doobie.Update0
+import doobie.implicits.*
+import doobie.util.fragment.Fragment
+import zio.interop.catz.*
+
+/*
+ * When we created table ReportsExecution, we didn't use "timestamp with time zone" for column
+ * 'insertiondate' and 'compliancecomputationdate'
+ */
+class CheckTableReportsExecutionTz(
+    doobie: Doobie
+) extends BootstrapChecks {
+
+  import CheckTableReportsExecutionTz.*
+  import doobie.*
+
+  override def description: String =
+    "Check if database table  ReportsExecution has timestamps with time zone and correct it if not"
+
+  def addTz(table: String, column: String, overrideChange: Option[Fragment] = None): IOResult[Unit] = {
+
+    transactIOResult(s"Error when adding time zone to column ${table}.${column}")(xa =>
+      query(table, column, overrideChange).run.transact(xa)
+    ).unit
+  }
+
+  def migrationProg(table: String, overrideChange: Option[Fragment] = None): IOResult[Unit] = {
+    for {
+      _ <- addTz(table, insertionDateColumn, overrideChange)
+      _ <- addTz(table, complianceComputationDateColumn, overrideChange)
+    } yield ()
+  }
+
+  override def checks(): Unit = {
+    // Actually run the migration async to avoid blocking for that.
+    // There is no need to have it sync.
+    migrationProg(reportsExecution)
+      .catchAll(err => BootstrapLogger.error(s"Error when trying to add time zone to ReportsExecution table: ${err.fullMsg}"))
+      .forkDaemon
+      .runNow
+  }
+
+}
+
+object CheckTableReportsExecutionTz {
+  val reportsExecution:                String = "reportsexecution"
+  val insertionDateColumn:             String = "insertiondate"
+  val complianceComputationDateColumn: String = "compliancecomputationdate"
+
+  // overrideChange is only used for tests
+  def query(table: String, column: String, overrideChange: Option[Fragment] = None): Update0 = {
+    def toQuotedFr(s: String) = Fragment.const("'" + s + "'")
+    def toFr(s:       String) = Fragment.const(s)
+
+    val alter: Fragment = fr"ALTER TABLE " ++ toFr(table) ++ fr" ALTER COLUMN " ++ toFr(
+      column
+    ) ++ fr" TYPE TIMESTAMP WITH TIME ZONE USING " ++ toFr(column) ++ fr" AT TIME ZONE 'UTC';"
```

Comment:
I'm not sure for the 'UTC' timezone, in our [Doobie Meta](https://github.com/Normation/rudder/blob/branches/rudder/8.2/webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/db/Doobie.scala#L182) we write dates using no specific timezone and I thought the PG session current timezone is used (`SHOW TIMEZONE;`), so maybe 'UTC' is a special test value here ?
I could not test the behavior because I have no reports locally but I believe that if the timezone is something else than UTC we could end up assigning wrong values to the columns

Reply:
we don't any information about what was used to write these timezone, this is lost for ever. The server config can have changed. And since it's not really use, it doesn't not matter. UTC is a as correct value, as other and in case of problem it will be much more easy to guess what happened if we don't have to check for what was the timezone when the migration was done.